jetson/science_bridge/src/__main__.py

```diff
@@ -0,0 +1,391 @@
+        # const int8_t whiteLED = 5;
+        # const int8_t uvBulb = 6;
+        # const int8_t nichWire0 = 7;
+        # const int8_t nichWire1 = 8;
+        # const int8_t nichWire2 = 9;
+        # const int8_t ramanLaser = 10;
+        # Starts from 0-2 for all the leds
+        # Resets to 1 starting at Laser(3) so offset by 2
+        # if (translated_device > 2):
+        #     translated_device -= 2
+        message = message.format(device=translated_device,
+                                 enable=int(struct.enable))
+        print(message)
+        while(len(message) < 30):
+            message += ","
+        self.ser.close()
+        self.ser.open()
+        if self.ser.isOpen():
+            self.ser.write(bytes(message, encoding='utf-8'))
+        print("Mosfet Received")
+        pass
+
+    def nav_status(self, channel, msg):  # TODO - fix make better
+        print("Received nav req")
+        # Off, Done, Else
+
+        struct = NavStatus.decode(msg)
+        message = "$Mosfet,{device},{enable},11111111"  # TODO - PROBABLY OUTDATED, WRONG PADDING
+        # All Leds are 1 digit so hardcode in padding
+
+        # Off = Blue
+        if struct.nav_state_name == "Off":
+            print("navstatus off")
+            offmessage = message.format(device=2, enable=1) + "1"
+            self.ser.write(bytes(offmessage, encoding='utf-8'))
+            prev = 2
+        # Done = Flashing green
+        elif struct.nav_state_name == "Done":
+            print("navstatus Done")
+            # Flashing by turning on and off for 1 second intervals
+            # Maybe change to
+            for i in range(0, 6):  # TODO - CHANGE TO RANGE (6)
+                self.ser.write(bytes(message.format(device=1, enable=1), encoding='utf-8'))
+                time.sleep(1)
+                self.ser.write(bytes(message.format(device=1, enable=0), encoding='utf-8'))
+                time.sleep(1)
+                prev = 1
+        # Everytime else = Red
+        else:
+            print("navstatus else")
+            messageon = message.format(device=0, enable=1)
+            self.ser.write(bytes(messageon, encoding='utf-8'))
+            prev = 0
+        time.sleep(1)
+        # Green should be in a finished state so no need to turn it off
+        if (prev != 2):
+            self.ser.write(bytes(message.format(device=2, enable=0),
+                                 encoding='utf-8'))
+        if (prev != 0):
+            self.ser.write(bytes(message.format(device=0, enable=0),
+                                 encoding='utf-8'))
+
+    def servo_transmit(self, channel, msg):
+        # get cmd lcm and send to nucleo
+        struct = ServoCmd.decode(msg)
+        print("Received Servo Cmd")
+        # parse data into expected format
+        message = "$Servo,{angle0},{angle1},{angle2}"
+        message = message.format(angle0=struct.angle0, angle1=struct.angle1, angle2=struct.angle2)
+        print(message)
+        while(len(message) < 30):
```

Comment:
30 --> const of like UART_TRANSMIT_MSG_LEN or something in that vein?

Reply:
Yes